client/components/capture-authorization-button/test/index.test.tsx

```diff
@@ -0,0 +1,85 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import user from '@testing-library/user-event';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CaptureAuthorizationButton from '../';
+import { useAuthorization } from 'wcpay/data';
+import { Authorization } from 'wcpay/types/authorizations';
+
+jest.mock( 'wcpay/data', () => ( {
+	useAuthorization: jest.fn(),
+} ) );
+
+const mockUseAuthorization = useAuthorization as jest.MockedFunction<
+	typeof useAuthorization
+>;
+
+function renderCaptureAuthorizationButton(
+	orderId: number,
+	paymentIntentId: string
+) {
+	return render(
+		<CaptureAuthorizationButton
+			orderId={ orderId }
+			paymentIntentId={ paymentIntentId }
+		/>
+	);
+}
+
+describe( 'CaptureAuthorizationButton', () => {
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	test( 'should render normal status', () => {
+		mockUseAuthorization.mockReturnValue( {
+			doCaptureAuthorization: jest.fn(),
+			isLoading: false,
+			authorization: {} as Authorization,
+		} );
+		const { container } = renderCaptureAuthorizationButton(
+			42,
+			'paymentIntentId'
+		);
+
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'should render busy status', () => {
+		mockUseAuthorization.mockReturnValue( {
+			doCaptureAuthorization: jest.fn(),
+			isLoading: true,
+			authorization: {} as Authorization,
+		} );
+		const { container } = renderCaptureAuthorizationButton(
+			42,
+			'paymentIntentId'
+		);
+
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'should call capture function when clicked', async () => {
+		const doCaptureAuthorizationMock = jest.fn();
+		mockUseAuthorization.mockReturnValue( {
+			doCaptureAuthorization: doCaptureAuthorizationMock,
+			isLoading: false,
+			authorization: {} as Authorization,
+		} );
+
+		renderCaptureAuthorizationButton( 42, 'paymentIntentId' );
+
+		expect( doCaptureAuthorizationMock.mock.calls.length ).toBe( 0 );
+
+		await user.click( screen.getByRole( 'button' ) );
+
+		expect( doCaptureAuthorizationMock.mock.calls.length ).toBe( 1 );
```

Comment:
One extra thing here: Should we also test if the state of the button switches to `busy` after we click on it + after the capture function is called?

Reply:
I think it is a good idea. I made an attempt to test it in this commit [4adfb2f](https://github.com/Automattic/woocommerce-payments/pull/4960/commits/4adfb2f668e2dc3b52d181a1340ef99076006724). I am not fully convinced of the test though. The problem is that this component relies on the `useAuthorization` hook to populate the `isBusy`  prop of the button (via isLoading). So I need to change manually the `isLoading` prop and then rerender the component.  Please have a look and let me know what you think.

Reply:
I think the mock function you used is clear and communicates what we are trying to test. I think it's good to keep 👍